retail/interactive-tutorial/RetailProducts.Samples/SetInventorySample.cs

```diff
@@ -0,0 +1,138 @@
+    /// Get the retail product with inventory info.
+    /// </summary>
+    /// <param name="product">The actual product.</param>
+    /// <returns>Retail product with updated inventory info.</returns>
+    private static Product SetInventoryForProduct(Product product)
+    {
+        FulfillmentInfo newFulfillmentInfo = new FulfillmentInfo
+        {
+            Type = "pickup-in-store",
+            PlaceIds = { "store1", "store2" }
+        };
+
+        product.FulfillmentInfo.Clear();
+        product.FulfillmentInfo.Add(newFulfillmentInfo);
+        product.PriceInfo.Price = 15.0f;
+        product.PriceInfo.OriginalPrice = 20.0f;
+        product.PriceInfo.Cost = 8.0f;
+        product.PriceInfo.CurrencyCode = "USD";
+
+        return product;
+    }
+
+    /// <summary>
+    /// Get the set inventory request.
+    /// </summary>
+    /// <param name="productWithInventory">The actual product with inventory info.</param>
+    /// <returns>Set inventory request.</returns>
+    private static SetInventoryRequest GetSetInventoryRequest(Product productWithInventory)
+    {
+        // The request timestamp.
+        DateTime requestTimeStamp = DateTime.UtcNow;
+
+        SetInventoryRequest setInventoryRequest = new SetInventoryRequest
+        {
+            Inventory = productWithInventory,
+            SetTime = Timestamp.FromDateTime(requestTimeStamp),
+            AllowMissing = true,
+            SetMask = new FieldMask
+            {
+                Paths = { "price_info", "availability", "fulfillment_info", "available_quantity" }
+            }
+        };
+
+        Console.WriteLine("Set inventory request:");
+        Console.WriteLine($"Product name: {setInventoryRequest.Inventory.Name }");
+        Console.WriteLine($"Product fultillment info: {setInventoryRequest.Inventory.FulfillmentInfo}");
+        Console.WriteLine();
+
+        return setInventoryRequest;
+    }
+
+    /// <summary>
+    /// Call the Retail API to set product inventory
+    /// </summary>
+    /// <param name="product">The actual product.</param>
+    public static void SetProductInventory(Product product)
+    {
+        Product productWithInventory = SetInventoryForProduct(product);
+        SetInventoryRequest setInventoryRequest = GetSetInventoryRequest(productWithInventory);
+
+        ProductServiceClient client = ProductServiceClient.Create();
+
+        // Make the request.
+        Operation<SetInventoryResponse, SetInventoryMetadata> response = client.SetInventory(setInventoryRequest);
+
+        Console.WriteLine("The operation was started:");
+        Console.WriteLine(response.Name);
+        Console.WriteLine();
+
+        Console.WriteLine("Please wait till opeartion is done");
+        Console.WriteLine();
+
+        // Poll until the returned long-running operation is complete.
+        Operation<SetInventoryResponse, SetInventoryMetadata> setInventoryResult = response.PollUntilCompleted();
+
+        Thread.Sleep(60000);
```

Comment:
This really needs an explanation after having polled until completion.

I'll raise this internally.